packages/frames.js/src/next/getCurrentUrl.ts

```diff
@@ -0,0 +1,29 @@
+import { IncomingMessage } from "http";
+import { NextApiRequest } from "next";
+import type { NextRequest } from "next/server.js";
+
+export function getCurrentUrl(
+  req: Request | NextRequest | NextApiRequest | IncomingMessage
+): URL | undefined {
+  const scheme =
+    process.env.VERCEL_ENV === "production" ? "https://" : "http://";
+  const vercelUrl = process.env.VERCEL_URL;
```

Comment:
what happens when you deploy outside vercel? it's just ignored?

Reply:
it is ignored in that case

Reply:
But yes, not that great tbh. Could be treated as quick fix and then we could do a proper fix. That would require more extensive tests so we cover different scenarios.